resources/configurationFiles/bam2fastq.xml

```diff
@@ -1,100 +1,122 @@
-            <output type="filegroup" fileclass="TextFile" scriptparameter='FILENAME_LANEFILES' check="true" passas="array"/>
+            <input type="file" typeof="BamFile" scriptparameter='FILENAME_BAM'/>
+            <output type="filegroup" fileclass="TextFile" scriptparameter='FILENAME_UNSORTED_FASTQ' check="true" passas="array"/>
         </tool>
-        <tool name='sortFastqSingle' value='sortFastqSingle.sh' basepath="bam2fastq">
+
+        <tool name='sortFastqSingle' value='coreutilsSortFastqSingle.sh' basepath="bam2fastq">
             <resourcesets>
-                <rset size="t" memory="0.1" cores="1" nodes="1" walltime="00:10:00"/>
-                <rset size="l" memory="5G" cores="8" nodes="1" walltime="10:00:00"/>
+                <rset size="t" memory="0.1" cores="8" nodes="1" walltime="00:10:00"/>
+                <rset size="m" memory="10G" cores="8" nodes="1" walltime="10:00:00"/>
+                <rset size="l" memory="10G" cores="8" nodes="1" walltime="50:00:00"/>
+                <rset size="xl" memory="10G" cores="8" nodes="1" walltime="100:00:00"/>
             </resourcesets>
             <input type="file" typeof="TextFile" scriptparameter='FILENAME_FASTQ'/>
+            <input type="string" setby="callingCode" scriptparameter='readGroup'/>
             <output type="file" typeof="TextFile" scriptparameter='FILENAME_SORTED_FASTQ' check="true"/>
         </tool>
-        <tool name='sortFastqPair' value='sortFastqPair.sh' basepath="bam2fastq">
+
+        <tool name='sortFastqPair' value='coreutilsSortFastqPair.sh' basepath="bam2fastq">
             <resourcesets>
-                <rset size="t" memory="0.1" cores="1" nodes="1" walltime="00:10:00"/>
-                <rset size="l" memory="10G" cores="8" nodes="1" walltime="10:00:00"/>
+                <rset size="t"  memory="0.1" cores="8" nodes="1" walltime="00:10:00"/>
+                <rset size="m"  memory="10G" cores="8" nodes="1" walltime="10:00:00"/>
+                <rset size="l"  memory="10G" cores="8" nodes="1" walltime="50:00:00"/>
+                <rset size="xl" memory="10G" cores="8" nodes="1" walltime="100:00:00"/>
             </resourcesets>
             <input type="file" typeof="TextFile" scriptparameter='FILENAME_FASTQ1'/>
             <input type="file" typeof="TextFile" scriptparameter='FILENAME_FASTQ2'/>
-            <input type="string" setby="callingCode" scriptparameter='READGROUP'/>
+            <input type="string" setby="callingCode" scriptparameter='readGroup'/>
             <output type="tuple">
                 <output type="file" typeof="TextFile" scriptparameter='FILENAME_SORTED_FASTQ1' check="true"
-                        filename='${fastqOutputDirectory}/${sourcefileAtomicPrefix,delimiter="."}_${READGROUP}_R1.sorted.fastq.gz'/>
+                        filename='${fastqOutputDirectory}/${cvalue,name=bamFileName}_fastqs/${cvalue,name=readGroup}/sequence/${sourcefileAtomicPrefix,delimiter="_R1"}_R1.sorted.fastq.gz'/>
```

Comment:
you do not need the cvalue syntax anymore. Just put in ${bamFileName} or ${readGroup}.